src/combinators/box_body.rs

```diff
@@ -0,0 +1,69 @@
+use crate::Body;
+use bytes::Buf;
+use std::{
+    fmt,
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+/// A boxed [`Body`] trait object.
+pub struct BoxBody<D> {
+    inner: Pin<
+        Box<
+            dyn Body<Data = D, Error = Box<dyn std::error::Error + Send + Sync>>
```

Comment:
Should this put a `Box<dyn Error>` in here, or should it be `BoxBody<D, E>`? I don't know the answer, but we probably should know why we pick one over the other.

Reply:
Hm true. On second thought I think `BoxBody<D, E>` is nicer. Users can always use `map_err` if they want boxed errors as well.